resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsLifecycleManager.scala

```diff
@@ -217,14 +217,42 @@ private[spark] class ExecutorPodsLifecycleManager(
     ExecutorExited(exitCode, exitCausedByApp, exitMessage)
   }
 
+  // A utility function to try and help people figure out whats gone wrong faster.
+  private def describeExitCode(code: Int): String = {
+    val humanStr = code match {
+      case 0 => "(success)"
+      case 1 => "(generic, look at logs to clarify)"
+      case 42 => "(douglas adams)"
+      // Spark specific
+      case 10 => "(Uncaught exception)"
+      case 50 => "(Uncaught exception)"
+      case 52 => "(JVM OOM)"
+      case 53 => "(DiskStore failed to create temp dir)"
+      // K8s & JVM specific exit codes
+      case 126 => "(not executable - possibly perm or arch)"
+      case 137 => "(SIGKILL, possible container OOM)"
+      case 139 => "(SIGSEGV: that's unexpected)"
+      case 255 => "(exit-1, your guess is as good as mine)"
+      case _ => ""
```

Comment:
For codes with no description, is there more context that can be given here?

Reply:
I'm not sure. Maybe we could say "check the logs" WDYT? Or we could generate a search link for "spark exit code number"

Reply:
Maybe we could pass through a message like this, `exited with code ${code}`

Reply:
Oh, I see that this is what’s happening as the return, more or less. I’m not sure in this case what I would out here. Is this an unknown code?

Reply:
Yeah unexpected is probably a good string here.